openedx/core/djangoapps/credit/tasks.py

```diff
@@ -33,23 +33,94 @@ def update_course_requirements(course_id):
+    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements, AttributeError) as exc:
         LOGGER.error('Error on adding the requirements for course %s - %s', course_id, unicode(exc))
         raise update_course_requirements.retry(args=[course_id], exc=exc)
     else:
         LOGGER.info('Requirements added for course %s', course_id)
 
 
-def get_min_grade_for_credit(course):
-    """ Returns the min_grade for the credit requirements """
-    return getattr(course, "min_grade", 0.8)
+def _get_course_credit_requirements(course):
+    """ Returns the list of credit requirements for the given course
+
+    It returns the minimum_grade_credit and also the ICRV checkpoints
+    if added any in the course
+
+    Args:
+        course(Course): The course object
+
+    Returns:
+        List of minimum_grade_credit and ICRV requirements
+    """
+    icrv_requirements = _get_credit_course_requirement_xblocks(course)
+    min_grade_requirement = _get_min_grade_requirement(course)
+    icrv_requirements.extend(min_grade_requirement)
+    return icrv_requirements
+
+
+def _get_min_grade_requirement(course):
+    """ Returns the list of minimum_grade_credit requirements for the given course
+
+    Args:
+        course(Course): The course object
+
+    Raises:
+        AttributeError if the course has not minimum_grade_credit attribute
+
+    Returns:
+        The list of minimum_grade_credit requirements
+    """
+    requirement = [
+        {
+            "namespace": "grade",
+            "name": "grade",
+            "criteria": {
+                "min_grade": getattr(course, "minimum_grade_credit")
+            }
+        }
+    ]
+    return requirement
+
+
+def _get_credit_course_requirement_xblocks(course):  # pylint: disable=invalid-name
+    """ Generates a course structure dictionary for the specified course.
+
+    Args:
+        course(Course): The course object
+
+    Returns:
+        The list of credit requirements xblocks dicts
+    """
+
+    blocks_stack = [course]
+    requirements_blocks = []
+    while blocks_stack:
+        curr_block = blocks_stack.pop()
+        children = curr_block.get_children() if curr_block.has_children else []
+        if _is_credit_requirement(curr_block):
+            block = {
+                "namespace": curr_block.get_credit_requirement_namespace(),
+                "name": curr_block.get_credit_requirement_name(),
+                "criteria": ""
+            }
+            requirements_blocks.append(block)
+
+        # Add this blocks children to the stack so that we can traverse them as well.
+        blocks_stack.extend(children)
+    return requirements_blocks
+
+
+def _is_credit_requirement(xblock):
+    """ Check if the given xblock is a credit requirement
+
+    Args:
+        xblock(XBlock): The given xblock object
+
+    Returns:
+        True if xblock is a credit requirement else False
+    """
+    if callable(getattr(xblock, "is_course_credit_requirement", None)):
```

Comment:
FYI: @chrisndodge this is the approach we're working on for discovering XBlocks in the course structure that count as credit requirements.